delta-app/src/main/java/io/cdap/delta/store/DraftService.java

```diff
@@ -290,14 +290,29 @@ private TableAssessmentResponse assessTable(SourceTable sourceTable, TableRegist
 
     // fetch detail about the table, then filter out columns that will not be read by the source
     TableDetail detail = tableRegistry.describeTable(db, table);
+    List<Problem> missingFeatures = new ArrayList<>(detail.getFeatures());
+    List<String> unselectedPrimaryKeys = detail.getPrimaryKey().stream()
+      // if there are no columns specified, it means this primary key will be read/selected
+      .filter(columnWhitelist.isEmpty() ? col -> false : col -> !columnWhitelist.contains(col))
+      .collect(Collectors.toList());
+    if (!unselectedPrimaryKeys.isEmpty()) {
+      missingFeatures.add(
+        new Problem("Missing Primary Key",
+                    String.format("Column(s) '%s' is(are) part of the primary key for table '%s' in database '%s', " +
```

Comment:
it's annoying to write the code, but it's a cleaner experience if only 'is' or 'are' are displayed in the error message, not both. Let's check for the size of unselectedPrimaryKeys and modify the message accordingly.

Reply:
Sure, we can definitely do that.